x-pack/plugins/reporting/public/share_context_menu/register_csv_reporting.tsx

```diff
@@ -55,7 +55,9 @@ export const reportingCsvShareProvider = ({
 
     // TODO: add abstractions in ExportTypeRegistry to use here?
     let capabilityHasCsvReporting = false;
-    if (usesUiCapabilities) {
+    if (!application.capabilities.discover?.save && !objectId) {
+      capabilityHasCsvReporting = false;
+    } else if (usesUiCapabilities) {
```

Comment:
It should be possible for users to create CSV reports from unsaved reports as well, as of #99313

It's also possible for a user to have permissions that grant privilege to generate reports, but not create saved searches.